### ISSUE TITLE
typing error

### ISSUE THREAD

**Issue by ksnyde** [amcharts/amcharts4#2272]
![image](https://user-images.githubusercontent.com/980757/77255947-46c48f80-6c28-11ea-85ad-df2efe44dc84.png)

The current typing for the `min` and `max` properties on an Axis is fixed as a **Number** when in fact it can/should be able to set to it's default value of _undefined_.

The correct typing would be:

```typescript
{
   min?: number
   max?: number
}
```

although because I suspect -- haven't checked source yet -- that this typing is coming implicitly from a class definitions public properties then you would instead type it as:

```typescript
{
  public min: number | undefined
}
```

I know that sometimes this kind of typing error comes as the result of the inconvenience downstream of a bunch of conditional logic which now is burdened with validating that the property is actually set to a value. This sometimes is a _good_ burden because it is in fact a real use case here but sometimes your business logic has already invalidated the `undefined` condition and I just pass something like this into the block which _knows_ that the property is set:

```typescript
function knowsItsDefined(klass: ValueAxis<any> & {min: number, max: number} { ... }
```

**Comment by ksnyde** [comment]
I tried to add to my comment but I guess it took too long ... here's a condensed comment ...

If you are defining interfaces then you'd want to define as:

```typescript
{
  min?: number,
  max?: number
}
```

but I suspect you're exporting the implicit typing that comes from the public interface of a class, so you'd want to type as:

```typescript
{
  min: number | undefined
  max: number | undefined
}
```

I typically see this mistake when people want to avoid the burden of a bunch of conditional logic in places where in fact they _know_ that the values have been set. This can be avoided though by simply passing something like the following to those blocks of code that truly do know that properties like min/max are set:

```typescript
function iKnowTheyreSet(klass: ValueAxis<any> && { min: number, max: number) { ... }
```

**Comment by ksnyde** [comment]
To get around this typing error I am forced in my code to do the following:

```typescript
axis.value.min = min === undefined ? ((undefined as unknown) as number) : Number(min)
 axis.value.max = max === undefined ? ((undefined as unknown) as number) : Number(max)
```

**Comment by Pauan** [comment]
@ksnyde As explained in our docs, [we do not currently support `strictNullChecks` with V4](https://www.amcharts.com/docs/v4/getting-started/using-typescript-or-es6/#TypeScript_strict_mode), since V4 was made years before `strictNullChecks` existed, and it is very difficult to migrate our existing code.

**Comment by ksnyde** [comment]
This is not a strictNullChecks issue, is it? What i mean is you can keep the `strictNullChecks` option off in your code base but but still type the option as optional.

**Comment by Pauan** [comment]
> we are using strict null checks and this is the only issue we've hit so far

I'd be very surprised if this is the only issue. We'll consider changing `ValueAxis`, but we don't really want to play whack-a-mole (one change leading to another change leading to another change...)

**Comment by ksnyde** [comment]
Sure I understand. I will hopefully be releasing an open source composable library to the VueJS community which will expose your awesome work in an easier fashion. My footprint at the moment to your library is currently about 40% and haven't run into other instances yet but once we're at 100% I'll try and get back to this thread and let you know.

**Comment by Pauan** [comment]
@ksnyde We changed `ValueAxis.min` and `ValueAxis.max` to accept `undefined`, the changes will be in the next release. We'll let you know when it's released.

**Comment by martynasma** [comment]
Fixed in [4.9.13](https://github.com/amcharts/amcharts4/releases/tag/4.9.13).

## [4.9.13] - 2020-04-05

### Added
- Global options setting added `am4core.options.suppressErrors` (default `false`). Set it to `true` to disable error-invoked modal display.

### Fixed
- Setting `locale` as string (e.g. `locale: "fr_FR"`) in JSON config was not working.
- `ValueAxis.min` and `ValueAxis.max` now accept `undefined`.
- `NumberFormatter` and `DateFormatter` will not longer critically fail if invalid `Intl` data is passed to them. Instead they will return `"Invalid"` string.
- Responsive: Sometimes objects were being revealed after coming back from responsive mode when thet shouldn't.
- `locations` setting of the `SeriesDataItem` were not being copied to grouped data items (when `DateAxis.groupData = true`).
- Shadow DOM: Made some changes to eliminate errors in IE11.
- Shadow DOM: `ExportMenu` and `Popup`/`Modal` CSS was not being properly applied causing those elements to lose styling and positioning.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
